ngraph/frontend/paddlepaddle/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "paddlepaddle_ngraph_frontend")
+
+file(GLOB_RECURSE LIBRARY_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cc)
+file(GLOB_RECURSE LIBRARY_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/src/*.hpp ${CMAKE_CURRENT_SOURCE_DIR}/src/*.h)
+file(GLOB_RECURSE LIBRARY_PUBLIC_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/*.hpp)
+
+if (TARGET ext_protobuf)
+#    add_dependencies(ngraph::paddlepaddle_ngraph_frontend ext_protobuf)
+    add_dependencies(${TARGET_NAME} ext_protobuf)
+endif()
+
+set(${TARGET_NAME}_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+# Create named folders for the sources within the .vcproj
+# Empty name lists them directly under the .vcproj
+
+source_group("src" FILES ${LIBRARY_SRC})
+source_group("include" FILES ${LIBRARY_HEADERS})
+source_group("public include" FILES ${LIBRARY_PUBLIC_HEADERS})
+
+set(PROTO_SRCS)
+set(PROTO_HDRS)
+
+# Generate protobuf file on build time for each '.proto' file in src/proto
+file(GLOB proto_files ${CMAKE_CURRENT_SOURCE_DIR}/src/proto/*.proto)
+
+if(CMAKE_CROSSCOMPILING)
+    set(PDPD_PROTOC_EXECUTABLE ${SYSTEM_PROTOC})
+else()
+    set(PDPD_PROTOC_EXECUTABLE $<TARGET_FILE:protoc>)
+endif()
+
+message("PDPD_PROTOC_EXECUTABLE is [${PDPD_PROTOC_EXECUTABLE}]")
+
+foreach(INFILE ${proto_files})
+    get_filename_component(FILE_DIR ${INFILE} DIRECTORY)
+    get_filename_component(FILE_WE ${INFILE} NAME_WE)
+    set(OUTPUT_PB_SRC ${CMAKE_CURRENT_BINARY_DIR}/${FILE_WE}.pb.cc)
+    set(OUTPUT_PB_HEADER ${CMAKE_CURRENT_BINARY_DIR}/${FILE_WE}.pb.h)
+    set(GENERATED_PROTO ${INFILE})
+    add_custom_command(
+            OUTPUT "${OUTPUT_PB_SRC}" "${OUTPUT_PB_HEADER}"
+            COMMAND ${PDPD_PROTOC_EXECUTABLE} ARGS --cpp_out ${CMAKE_CURRENT_BINARY_DIR} -I ${FILE_DIR} ${FILE_WE}.proto
+            DEPENDS ${PDPD_PROTOC_EXECUTABLE} ${GENERATED_PROTO}
+            COMMENT "Running C++ protocol buffer compiler (${PDPD_PROTOC_EXECUTABLE}) on ${GENERATED_PROTO}"
+            VERBATIM)
+    list(APPEND PROTO_SRCS "${OUTPUT_PB_SRC}")
+    list(APPEND PROTO_HDRS "${OUTPUT_PB_HEADER}")
+endforeach()
+
+add_custom_target(${TARGET_NAME}_proto DEPENDS ${PROTO_SRCS} ${PROTO_HDRS})
+
+set_source_files_properties(${PROTO_SRCS} ${PROTO_HDRS} PROPERTIES GENERATED TRUE)
+
+# Disable all warnings for generated code
+set_source_files_properties(${PROTO_SRCS} ${PROTO_HDRS} PROPERTIES COMPILE_FLAGS -w)
+
+include_directories(${Protobuf_INCLUDE_DIRS} ${${TARGET_NAME}_INCLUDE_DIR})
+
+# Create shared library
+add_library(${TARGET_NAME} SHARED  ${LIBRARY_SRC} ${LIBRARY_HEADERS} ${LIBRARY_PUBLIC_HEADERS} ${PROTO_SRCS} ${PROTO_HDRS})
```

Comment:
I suppose we need to use `MODULE` instead of `SHARED`? Since it's a plugin?

Reply:
We discussed internally and we'll continue using of "SHARED" library approach - integration with frameworks (PaddlePaddle) in our case will require work with concrete PDPDInputModel etc. At least it will be needed to define 'extensions' in future